File: manga_be/src/main/java/com/reader/manga/domain/valueobjects/screens/listing/animes/AnimeListingVO.java
package com.reader.manga.domain.valueobjects.screens.listing.animes;

import com.reader.manga.domain.enums.TagType;
import lombok.Builder;

import java.time.LocalDate;
import java.util.List;

/**
 * VO para centralizar dados a ser mandado para tela de listagem de animes.
 * @author andrewgo
 */
@Builder
public record AnimeListingVO(
        Long idAnime,
        Integer note,
        String uriImage,
        String titleAnime,
        LocalDate launchYear,
        Boolean isFavorite,
        List<TagType> tags,
        List<EpisodeToAnimesVO> episodes
) {}

File: manga_fe/src/@types/AnimeListingVO.ts
import type { EpisodeToAnimesVO } from "./iEpisodeToAnimesVO"

export interface AnimeListingVO {
  idAnime: number
  note: number
  uriImage: string
  titleAnime: string
  launchYear: Date
  isFavorite: boolean
  tags: Array<string>
  episodes: Array<EpisodeToAnimesVO>
}